aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,291 @@
+	:language: c#
+	:linenos:
+	:emphasize-lines: 7
+
+We also implement all of these interfaces using a single class, ``Operation``, that simply accepts a ``Guid`` in its constructor, or uses a new ``Guid`` if none is provided..
+
+Next, in ``ConfigureServices``, we add each type to the container according to its named lifetime:
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Startup.cs
+	:language: c#
+	:lines: 83-88
+	:linenos:
+	:dedent: 8
+
+Note that the *instance* lifetime type has been added with a known ID of ``Guid.Empty`` so it will be clear when this type is in use. We have also registered an ``OperationService`` that depends on each of the other ``Operation`` types, so that it will be clear within a request whether this service is getting the same instance as the controller, or a new one, for each operation type.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Services/OperationService.cs
+	:language: c#
+	:linenos:
+
+To demonstrate the object lifetimes within and between separate individual requests to the application, the sample includes an ``OperationsController`` that requests each kind of ``IOperation`` type as well as an ``OperationService``. The ``Index`` action then displays all of the controller's and service's ``OperationId`` values.
+
+.. literalinclude:: dependency-injection/sample/src/DependencyInjectionSample/Controllers/OperationsController.cs
+	:language: c#
+	:linenos:
+
+Now two separate requests are made to this controller action:
+
+.. image:: dependency-injection/_static/lifetimes_request1.png
+.. image:: dependency-injection/_static/lifetimes_request2.png
+
+Observe which of the ``OperationId`` values varies within a request, and between requests.
+
+- *Transient* objects are always different; a new instance is provided to every controller and every service.
+- *Scoped* objects are the same within a request, but different across different requests
+- *Singleton* objects are the same for every object and every request
+- *Instance* objects are the same for every object and every request, and are the exact instance that was specified in ``ConfigureServices``
+
+Request Services and Application Services
+-----------------------------------------
+
+The services available within an ASP.NET request from ``HttpContext`` fall into two collections: *ApplicationServices* and *RequestServices*.
+
+.. image:: dependency-injection/_static/application_request_services.png
+
+Request Services represent the services you configure and request as part of your application. These are the services that are available on a per-request basis. Application Services are limited to those things that are available on application startup. These are services that are outside the scope of one request, such as ``IHostingEnvironment``. When your objects specify dependencies, these are satisfied by the types found in ``RequestServices``, not ``ApplicationServices``.
```

Comment:
The ApplicationServices and RequestServices properties are [technically different](https://github.com/aspnet/Hosting/blob/9a12085ef95bc5d9ab4f78aa3c2e5ab88f927816/src/Microsoft.AspNet.Hosting/Internal/RequestServicesContainerFeature.cs). Both have always contained the exact same services. Only the lifetime of the resolved services are different.

I think attempting to resolve a scoped service from ApplicationServices could be a common pitfall, so I think we should make it clear constructor injection is preferred way to resolve services, particularly for controllers and the Startup class.

Inside of a controller, I can imagine use cases for using the RequestServices property, but I don't think there is any reason to use ApplicationServices. I'm not entirely sure why ApplicationServices is a property on HttpContext.